homeassistant/components/vallox/sensor.py

```diff
@@ -219,16 +223,10 @@ class ValloxSensorEntityDescription(SensorEntityDescription):
 )
 
 
-async def async_setup_platform(
-    hass: HomeAssistant,
-    config: ConfigType,
-    async_add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up the sensors."""
-    if discovery_info is None:
-        return
-
     name = hass.data[DOMAIN]["name"]
     coordinator = hass.data[DOMAIN]["coordinator"]
```

Comment:
The data needs to be stored as `hass.data[DOMAIN][entry.entry_id]` now since they can have multiple entries.

Reply:
Ok 👍 So if I understand it correctly, https://github.com/home-assistant/core/pull/62780#pullrequestreview-840109304 fixes the storage part of it, and then I need to change these two lines to the following? And obviously similarly in `fan.py`.

```suggestion
    name = hass.data[DOMAIN][entry.entry_id]["name"]
    coordinator = hass.data[DOMAIN][entry.entry_id]["coordinator"]
```

Reply:
Yes

Reply:
Fixed and pushed 👍